### PR TITLE
test: address MEDIUM and LOW test coverage gaps

### PR DESCRIPTION
**Pull request by alwynpan**
## Summary

- Add 40 new tests across all 4 suites to close remaining MEDIUM and LOW coverage gaps
- Total test count: 1397 → 1437 (+40 tests)
- Completes the full test quality audit (CRITICAL → HIGH → MEDIUM → LOW)

### Backend unit (+9)
- Model error propagation: User.create, User.updatePassword, User.findByIds, Role.create, Group.getExportMappings
- Role.create: duplicate-name constraint (23505) test
- Group.bulkCreate: empty array input
- GET /groups/:id: invalid UUID returns 400
- POST /groups: Group.findByName DB error returns 500

### Integration (+13)
- GET /users?groupId: filter by group, ungrouped users, invalid UUID
- POST /auth/register: registration disabled returns 403
- POST /users/import: within-batch duplicate email detection, invalid conflictAction
- POST /groups/import-mappings: skip-action rows in response
- GET /auth/me: deleted user JWT returns 401
- PUT /users/:id: admin role change verified via GET
- POST /groups/:id/join and /leave: invalid UUID returns 400
- PUT /users/:id/password: admin cannot change another user's password

### Frontend unit (+15)
- AuthContext: registrationEnabled flag, login error status field
- Dashboard: assignment_manager role rendering
- Settings: success message auto-dismiss timeout
- SetPassword: invalid token renders form (not "Invalid Link")
- CsvDropzone: drag-and-drop file handling
- **NEW** formatting.test.js: formatRoleName with all roles + edge cases
- **NEW** api.test.js: Axios interceptor auth header attach/omit

### E2E (+3)
- Unauthenticated /settings and /users/import redirect to /login
- Forgot-password empty email validation

## Test plan

- [x] Backend unit: 622 passed
- [x] Frontend unit: 548 passed
- [x] Integration: 171 passed
- [x] E2E: 96 passed
- [x] Lint: clean
- [x] Format: clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: frontend/tests/unit/utils/api.test.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    // Axios interceptors are stored in api.interceptors.request.handlers
+    const interceptor = api.interceptors.request.handlers[0];
+    const result = interceptor.fulfilled(config);
+
+    expect(result.headers.Authorization).toBe('Bearer test-jwt-token');
+  });
+
+  it('does not attach Authorization header when no token in localStorage', () => {
+    localStorage.removeItem('token');
+
```
[the remark this answers is not in the document]
Comment (alwynpan) [reply frontend/tests/unit/utils/api.test.js]: Fixed in 7b120b8 — added `beforeEach`/`afterEach` that clears `localStorage` and restores the original adapter, so tests are fully isolated.

File: frontend/tests/unit/utils/api.test.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    const config = { headers: {} };
+    const interceptor = api.interceptors.request.handlers[0];
+    const result = interceptor.fulfilled(config);
+
+    expect(result.headers.Authorization).toBeUndefined();
```
[the remark this answers is not in the document]
Comment (alwynpan) [reply frontend/tests/unit/utils/api.test.js]: Fixed in 7b120b8 — replaced internal `handlers[0].fulfilled` access with a mock adapter (`jest.fn()`) injected via `api.defaults.adapter`. Tests now assert on the config passed to the adapter during a real `api.get()` call, making them resilient to Axios internal changes.
